covid19-etl/etl/idph_vaccine.py

```diff
@@ -0,0 +1,268 @@
+import os
+import json
+
+import requests
+
+from etl.idph import IDPH
+from utils.format_helper import (
+    idph_get_date,
+    derived_submitter_id,
+)
+from utils.metadata_helper import MetadataHelper
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+ROOT_URL = "https://idph.illinois.gov/DPHPublicInformation/api/covidVaccine/getVaccineAdministrationCurrent"
+COUNTY_COVID_LINK_FORMAT = "https://idph.illinois.gov/DPHPublicInformation/api/covidVaccine/getVaccineAdministration?countyName={}"
+COUNTY_DEMO_LINK_FORMAT = "https://idph.illinois.gov/DPHPublicInformation/api/covidvaccine/getVaccineAdministrationDemos?countyname={}"
+TOTAL_VACCINE_LINK = "https://idph.illinois.gov/DPHPublicInformation/api/covidvaccine/getStatewideVaccineDetails"
+
+
+class IDPH_VACCINE(IDPH):
+    def __init__(self, base_url, access_token, s3_bucket):
+        super().__init__(base_url, access_token, s3_bucket)
+
+        self.program_name = "open"
+        self.project_code = "IDPH-Vaccine"
+        self.metadata_helper = MetadataHelper(
+            base_url=self.base_url,
+            program_name=self.program_name,
+            project_code=self.project_code,
+            access_token=access_token,
+        )
+        self.country = "US"
+        self.state = "IL"
+        self.date = ""
+        self.counties_inventory = {}
+
+        self.summary_locations = {}
+        self.summary_clinicals = {}
+        self.summary_group_demographic = {}
+
+    def parse_list_of_counties(self):
+        response = requests.get(ROOT_URL, headers={"content-type": "json"})
+        json_response = json.loads(response.text)
+        self.date = idph_get_date(json_response.get("lastUpdatedDate"))
+        root_json = json_response.get("VaccineAdministration")
+        counties = []
+        if root_json is None:
+            return counties
+        for item in root_json:
+            county = item.get("CountyName")
+            if county.lower() == "out of state":
+                continue
```

Comment:
did you confirm with the data ingestion team that we don't want to bring this data in?